unit-tests/dds/test-device-discovery.py

```diff
@@ -203,6 +203,9 @@ def __exit__( self, type, value, traceback ):
             reader_2 = dds.topic_reader( device_info_topic )
             reader_2.run( dds.topic_reader.qos() )
         test.check_equal( len(broadcast_devices), 2 )
+        # Add short sleep to avoid a possible deadlock. Devices broadcast is handled by `on_device_info_available` callback,
+        # we may still be checking for messages (in eProcima reader thread) when trying to delete reader_2.
+        sleep( 0.1 )
```

Comment:
Is this an issue in the test or in an actual use case?
Do we need to follow up with a full solution in the SDK?

Reply:
Note, this is testing the `realdds` library, not the `librealsense` entities.
The issue happens when a `dds_reader` is being deleted by one thread while checking if a message have been received in another. This is a python wrapper issue - `on_data_available` callback is protected by GIL, `del` also calls GIL internally.
SDK uses `dds_reader_thread` internally that is not exposed on the python wrapper.

Theoretically there could be issues for a user when callback functions are being handled while the object is being deleted from another thread, but I think it's the user responsibility to protect this use case.